spec/lib/rmt/config_spec.rb

```diff
@@ -0,0 +1,40 @@
+require 'rails_helper'
+
+RSpec.describe RMT::Config do
+  describe '#mirroring dedup_method' do
+    context 'handles nil' do
+      before { deduplication_method(nil) }
+      it('defaults to hardlink') { expect(described_class.deduplication_by_hardlink?).to be_truthy }
+    end
+
+    context 'handles empty string' do
+      before { deduplication_method('') }
+      it('defaults to hardlink') { expect(described_class.deduplication_by_hardlink?).to be_truthy }
+    end
+
+    context 'handles everything' do
+      before { deduplication_method(Class.new) }
+      it('defaults to hardlink') { expect(described_class.deduplication_by_hardlink?).to be_truthy }
```

Comment:
But is it really supposed to handle "everything"?

`railsconfig` gem actually supports [config schema validation](https://github.com/railsconfig/config#validation), so rather then testing weird things like `Class.new`, maybe let's add validation instead. Because `Class.new` is not a valid input.

Reply:
everything is an object in ruby

Reply:
Sure, but we expect only very specific object with only two very specific values. :-)
Everything else is invalid.